src/include/OpenImageIO/strutil.h

```diff
@@ -209,6 +209,26 @@ std::string OIIO_API join (const std::vector<std::string> &seq,
 
 
 
+// Helper template to test if a string is a generic type
+template<typename T>
+inline bool string_is (string_view s) {
+    return false; // Generic: assume there is an explicit specialization
+}
+// Special case for int
+template <> inline bool string_is<int> (string_view s) {
+    char *endptr = 0;
+    strtol (s.c_str(), &endptr, 10);
```

Comment:
I think that in this case you want to use data(), not c_str().

So... because a string_view is not necessarily null-terminated, but c_str() must return a valid null-terminated C string, the way c_str() works is tricky. If the character immediately following the string_view region is a null, then c_str just returns data(), piece of cake. But if it doesn't -- that is, if the view is an interior section of another string -- then in order to safely return a NULL string that the caller doesn't have to free, it actually creates a ustring (which can safely reside in the ustring table for the duration of the process execution) and returns the ustring's c_str(). 

So the bottom line is that if you think you'll call this on the interior of strings frequently, c_str() will be expensive and wasteful, creating lots of "permanent" ustring table entries needlessly. Using data() will avoid this. The only down-side is that it may end up reading past the end of the view. But that should be ok, because it's still guaranteed to eventually be null-terminated.